src/main/java/GameField.java

```diff
@@ -56,7 +56,7 @@ private boolean isOutOfGameField(Rectangle objectBody){
     private boolean canMove(GameObject gameObject){
         Point speed = gameObject.getSpeed();
         
-        movedBody.setRectangle(gameObject.getBody());
+        movedBody.copyRectangle(gameObject.getBody());
```

Comment:
You've already have copy constructor:

Rectangle(Rectangle rectangle); 

use it